### DISCUSSION TITLE
tRPC async mutations

### DISCUSSION THREAD

**Discussion by ahriannn** [react-hook-form/react-hook-form#9989] [Q&A] [answered]
Hi I was wondering how people are using tRPC with onSubmit handling because the mutations can handle onError events which then causes Unhandled Runtime Error, the only workaround is to wrap the function in a try-catch with an empty catch.
```
  const onSubmit: SubmitHandler<FormSchemaType> = async (data) => {
    await submitData(data);
  };
```

**Comment by leapful** [comment, marked as answer]
@zokins 

`useMutation` hook from tRPC is a wrapper of react-query therefore you can use method `mutateAsync` with try catch statement inside `onSubmit` callback like a normal flow.

**Reply by ahriannn** [reply]
> @zokins
> 
> `useMutation` hook from tRPC is a wrapper of react-query therefore you can use method `mutateAsync` with try catch statement inside `onSubmit` callback like a normal flow.

I got it working with try/catch however wasn't ideal just because of the return type safety in the onError hook:
```
const onSubmit: SubmitHandler<FormSchemaType> = async (data) => {
    try {
      await mutate(data);
    } catch (error) {
      if (error instanceof TRPCClientError) {
        //Handle Error
      }
    }
}
```
